dashboard/app/models/school_info.rb

```diff
@@ -83,11 +93,12 @@ def school_state=(input)
   #
   # This method reports errors if the record has a country and is invalid.
   def validate_with_country
-    return unless country
+    return unless country && should_validate?
     country == 'US' ? validate_us : validate_non_us
   end
 
   def validate_non_us
+    return unless should_validate?
```

Comment:
This method seems to be only called from `validate_with_country`, where we already asserted `should_validate`. That said, I'm fine with the redundancy, if it makes sense.